packages/eslint-plugin/src/rules/require-array-sort-compare.ts

```diff
@@ -27,26 +27,16 @@ export default util.createRule({
 
     return {
       ":matches(CallExpression, OptionalCallExpression)[arguments.length=0] > :matches(MemberExpression, OptionalMemberExpression)[property.name='sort'][computed=false]"(
-        node: TSESTree.MemberExpression | TSESTree.OptionalMemberExpression,
+        callee: TSESTree.MemberExpression | TSESTree.OptionalMemberExpression,
       ): void {
-        // Get the symbol of the `sort` method.
-        const tsNode = service.esTreeNodeToTSNodeMap.get(node);
-        const sortSymbol = checker.getSymbolAtLocation(tsNode);
-        if (sortSymbol == null) {
-          return;
-        }
+        const tsNode = service.esTreeNodeToTSNodeMap.get(callee.object);
+        const calleeObjType = util.getConstrainedTypeAtLocation(
+          checker,
+          tsNode,
+        );
 
-        // Check the owner type of the `sort` method.
-        for (const methodDecl of sortSymbol.declarations) {
-          const typeDecl = methodDecl.parent;
-          if (
-            ts.isInterfaceDeclaration(typeDecl) &&
-            ts.isSourceFile(typeDecl.parent) &&
-            typeDecl.name.escapedText === 'Array'
-          ) {
-            context.report({ node: node.parent!, messageId: 'requireCompare' });
-            return;
-          }
+        if (util.isTypeArrayTypeOrUnionOfArrayTypes(calleeObjType, checker)) {
```

Comment:
I just realised that this should also mean that this rule now supports bracket accessed `sort` too - I'll check a test in.